gk/main.c

```diff
@@ -534,13 +609,10 @@ gk_proc(void *arg)
 				/* Drop non-IP packets. */
 				drop_packet(pkt);
 				continue;
-			} else if (pkt_is_nd(&packet, &gk_conf->net->front)) {
-				/*
-				 * TODO Use DPDK packet classification
-				 * and distribution here instead.
-				 */
-				if (submit_nd(pkt, &gk_conf->net->front) == -1)
-					drop_packet(pkt);
+			} else if (packet.flow.proto == ETHER_TYPE_IPv6 &&
+				     ipv6_if_configured(&gk_conf->net->front)) {
```

Comment:
It's okay to push this to `process_pkts_ipv6_acl()`, but I was thinking that if IPv6 is not configured, the ACLs will be empty, so they will free any packet that is pushed into classification. That is, the test if IPv6 is configured is redundant; it may give a small speedup at best.